Shared/ThresholdKeyView.swift

```diff
@@ -329,34 +371,44 @@ struct ThresholdKeyView: View {
                                     alertContent = "Password share already exists"
                                     showAlert = true
                                 }
+                                isLoadingAddPassword = false
                             }
                         }) {
                             Text("")
                         }.alert(isPresented: $showAlert) {
                             Alert(title: Text("Alert"), message: Text(alertContent), dismissButton: .default(Text("Ok")))
                         }
-                    }
+                    }.disabled(isLoadingAddPassword)
+                        .opacity(isLoadingAddPassword ? 0.5 : 1)
 
                     HStack {
                         Text("Change password")
                         Spacer()
+                        if isLoadingChangePassword {
+                            LoaderView()
+                        }
                         Button(action: {
-                            Task {
-                                let question = "what's your password?"
-                                let answer = "blublublu"
-                                _ = try! await SecurityQuestionModule.change_question_and_answer(threshold_key: threshold_key, questions: question, answer: answer)
-                                let key_details = try! threshold_key.get_key_details()
-                                totalShares = Int(key_details.total_shares)
-                                threshold = Int(key_details.threshold)
-
-                                alertContent = "Password changed to: \(answer)"
-                                showAlert = true
-                            }
-                        }) {
+                        Task {
+                            showInputPasswordAlert.toggle()
+                        }
+                        }){
                             Text("")
-                        }.alert(isPresented: $showAlert) {
+                        }.alert("Input Password", isPresented: $showInputPasswordAlert) {
+                            SecureField("New Password", text: $password)
+                            Button("Save", action: {
+                                Task {
+                                    await changeQnA()
```

Comment:
Do we need to wrap this into a separate function? For the example purposes it would probably be clearer to be expanded here, I don't see us using this anywhere else?